vm/systemSmartContracts/delegation.go

```diff
@@ -1968,6 +2042,48 @@ func (d *delegation) saveGlobalFundData(globalFundData *GlobalFundData) error {
 	return nil
 }
 
+func (d *delegation) checkOwnerInitialFunds(delegationConfig *DelegationConfig, caller []byte, callValue *big.Int) error {
+	// initial owner funds must be 0 or higher than min deposit
+	if delegationConfig.InitialOwnerFunds.Cmp(zero) > 0 {
+		return nil
+	}
+
+	if !d.isOwner(caller) {
+		return vm.ErrNotEnoughInitialOwnerFunds
+	}
+
+	minDeposit, err := d.getMinDeposit()
+	if err != nil {
+		return err
+	}
+	if minDeposit.Cmp(callValue) < 0 {
+		return fmt.Errorf("%w you must provide at least %s", vm.ErrNotEnoughInitialOwnerFunds, minDeposit.String())
+	}
+
+	delegationConfig.InitialOwnerFunds.Set(callValue)
+	err = d.saveDelegationContractConfig(delegationConfig)
```

Comment:
This should be removed: the function is called check and does a save.

Reply:
in extreme scenarios will get into this line. thus needs to save, and only here.

Reply:
The function is called check and we do a save. Let's rename the function then

Reply:
done